src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -494,14 +495,12 @@ export function sign(args: { signer: Account; transaction: AnyRawTransaction }):
  * @returns A SignedTransaction
  */
 export function generateSignedTransaction(args: InputSubmitTransactionData): Uint8Array {
-  const { transaction, senderAuthenticator, feePayerAuthenticator, additionalSignersAuthenticators } = args;
+  const { transaction, feePayerAuthenticator, additionalSignersAuthenticators } = args;
+  const senderAuthenticator = normalizeBundle(AccountAuthenticator, args.senderAuthenticator);
 
   const transactionToSubmit = deriveTransactionType(transaction);
 
-  if (
-    (feePayerAuthenticator || additionalSignersAuthenticators) &&
-    (transactionToSubmit instanceof MultiAgentRawTransaction || transactionToSubmit instanceof FeePayerRawTransaction)
-  ) {
+  if ("secondary_signer_addresses" in transactionToSubmit) {
```

Comment:
this is a hack that works because 
- feePayer has secondarySignerAddresses as an array of addresses or an empty array, so it "passes" this check
- multiAgent always has secondarySignerAddresses
right? 

Let's add a comment that this is a hack so one wont think and expect feePayerTransaction to always need secondarySignerAddresses

Reply:
Not really a hack, just looks hacky :(
Everything is strong typed and typescript correctly picks it up.
`transactionToSubmit` is `RawTransaction | FeePayerRawTransaction | MultiAgentRawTransaction`, and the last two always have that field defined.

Reply:
Futher cleaned it up and removed the unnecessary `deriveTransactionType` step